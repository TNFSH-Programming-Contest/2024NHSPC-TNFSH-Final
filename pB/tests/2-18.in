100 12
5 6 7 5 4 2 4 9 8 9 6 2 8 2 4 2 4 5 4 2 9 6 9 2 9 8 4 4 8 6 5 3 5 6 1 7 4 6 6 5 10 9 1 8 5 2 2 8 2 2 4 7 2 2 2 2 1 1 10 7 3 5 6 10 4 2 2 7 4 3 1 9 5 9 7 5 10 5 8 1 5 10 5 5 8 10 9 10 7 5 4 8 7 5 5 7 7 4 3 1
32 50 177415154 1275 78 1 18828123 4257901 135358 9 3 25
0 
9 4 9 8 2 7 1 5 6 3
11
0 
5 5 2 1 6 8
100
0 
10 1 8 10 6 3 4 11 12 2 9
66
0 
7 3 6 5 8 4 7 2
29
0 
0 
80
0 
3 2 11 3
100
0 
3 4 10 7
15
0 
7 5 3 6 7 8 1 11
21
0 
7 9 3 2 7 11 10 12
66
0 
4 11 9 7 10
24
0 
7 9 4 2 3 8 10 12
29
0 
9 3 8 10 2 7 6 1 9 4
68
0 
9 7 9 5 8 10 2 12 11 1
96
0 
2 1 3
16
0 
4 12 4 8 3
83
0 
4 1 6 10 2
84
0 
3 8 2 7
27
0 
10 6 3 1 7 10 4 5 2 8 12
41
0 
7 1 3 10 11 4 8 7
8
0 
1 5
59
0 
10 4 8 6 9 2 11 7 5 10 3
95
0 
6 4 10 11 1 12 6
81
0 
1 2
62
0 
6 11 1 3 12 4 8
87
0 
7 7 5 3 2 4 10 8
33
0 
1 8
100
0 
4 4 10 2 1
28
0 
5 4 9 3 7 5
82
0 
0 
38
0 
9 3 4 9 11 10 2 8 5 12
13
0 
6 1 9 4 2 6 8
93
0 
6 6 2 1 8 12 7
3
0 
2 5 6
83
0 
0 
34
0 
6 3 6 7 1 9 4
41
0 
2 12 11
22
0 
8 4 9 10 5 2 6 1 12
78
0 
10 1 5 11 10 8 4 7 6 2 12
73
0 
4 2 12 9 7
97
0 
3 12 10 6
77
0 
5 9 10 3 7 4
56
0 
7 7 11 8 4 10 1 9
37
0 
2 11 12
51
0 
5 7 8 4 6 1
9
0 
6 7 9 11 5 1 3
0
0 
4 3 8 2 6
87
0 
0 
46
0 
2 1 12
91
0 
9 9 4 1 2 7 10 5 12 8
51
0 
8 9 3 12 1 5 10 4 6
99
0 
2 10 7
64
0 
9 6 8 9 10 5 2 3 4 7
25
0 
8 5 6 4 9 7 12 2 1
68
0 
4 3 10 1 6
38
0 
9 11 8 12 2 6 10 1 4 9
2
0 
9 12 11 5 10 8 2 3 4 7
27
0 
6 5 2 6 3 12 7
28
0 
6 8 11 10 9 1 12
11
0 
8 8 4 1 9 5 7 11 6
34
0 
10 11 5 7 3 1 8 10 6 4 2
55
0 
9 3 7 1 12 8 9 2 5 10
56
0 
2 6 5
49
0 
5 3 8 1 11 12
77
0 
2 10 8
75
0 
1 12
76
0 
3 7 5 3
37
0 
1 9
11
0 
3 11 6 2
87
0 
6 5 2 6 11 1 12
27
0 
6 1 5 2 11 7 4
80
0 
2 10 6
75
0 
0 
94
0 
3 7 10 5
10
0 
6 7 1 9 12 8 4
74
0 
10 1 3 2 12 8 11 6 9 10 7
20
0 
4 6 2 7 5
94
0 
2 7 10
59
0 
9 11 5 1 8 7 6 2 9 10
7
0 
7 2 5 9 10 3 4 6
58
0 
10 10 11 7 12 5 9 1 8 4 2
45
0 
10 1 9 7 8 3 12 4 2 10 6
53
0 
10 7 10 4 6 8 1 9 11 5 3
89
0 
10 7 11 6 4 5 2 1 10 12 9
56
0 
10 11 12 9 7 6 3 4 1 8 10
43
0 
0 
95
0 
1 10
45
0 
9 5 3 7 11 10 4 12 2 8
2
0 
8 7 10 2 9 1 8 4 6
35
0 
8 5 7 10 6 1 12 8 3
71
0 
1 2
89
0 
6 7 5 4 9 12 2
41
0 
3 5 10 7
91
0 
7 8 1 11 3 6 4 10
98
0 
5 2 4 10 3 1
95
0 
4 9 8 6 11
90
0 
2 9 11
25
0 
2 3 5
32
0 
8 4 8 3 12 7 6 10 5
89
0 
0 
100
0 
2 3 6
55
0 
10 10 1 3 8 4 11 5 7 2 12
12
0 
7 1 6 3 7 4 11 5
69
0 
2 6 8
76
0 
1 9
74
0 
1 8
42
0 
4 3 11 6 7
78
0 
8 6 5 10 7 3 1 12 2
47
0 
9 7 12 2 8 10 5 3 4 9
15
0 
5 3 11 9 1 5
63
0 
9 11 8 5 7 2 12 1 9 10
55
0 
4 7 5 4 11
22
0 
6 3 11 1 6 2 4
50
0 
2 2 7
42
0 
8 11 6 2 1 5 3 9 7
92
0 
2 9 12
54
0 
8 8 5 7 6 3 1 12 10
36
0 
0 
84
0 
0 
84
0 
6 6 12 1 8 11 5
65
0 
3 7 1 4
70
0 
10 12 3 5 10 7 4 9 1 8 2
16
0 
7 7 2 5 6 11 1 4
81
0 
10 5 9 3 8 11 10 2 4 6 7
5
0 
0 
87
0 
9 11 1 4 12 7 10 9 6 3
97
0 
1 3
20
0 
7 1 6 9 3 10 7 5
88
0 
0 
12
0 
4 6 5 2 8
47
0 
3 11 7 4
6
0 
5 6 10 4 1 3
9
0 
1 9
36
0 
6 12 3 1 2 8 9
86
0 
8 8 11 9 6 5 12 7 1
15
0 
10 12 5 3 10 2 9 11 8 6 4
70
0 
5 3 4 9 5 11
94
0 
10 2 1 5 7 9 12 3 8 11 4
13
0 
6 8 2 5 7 9 1
51
0 
2 3 10
22
0 
9 10 8 6 12 11 2 1 4 9
20
0 
2 5 8
37
0 
5 3 7 4 12 1
32
0 
2 10 2
52
0 
4 9 10 3 1
66
0 
7 5 11 12 4 10 7 6
9
0 
3 7 2 9
73
0 
10 2 11 1 8 4 6 12 7 5 10
29
0 
1 1
24
0 
4 9 6 10 4
75
0 
9 2 3 10 6 11 1 8 7 9
53
0 
10 6 3 11 12 2 1 7 5 9 8
53
0 
6 1 11 9 2 6 12
80
0 
8 5 12 11 8 4 1 3 6
15
0 
9 9 7 6 8 3 5 12 10 1
25
0 
5 10 3 6 11 8
13
0 
2 1 2
63
0 
6 5 3 7 12 4 8
74
0 
6 12 6 2 9 11 4
48
0 
7 9 8 11 12 6 7 1
88
0 
5 11 10 1 3 7
8
0 
1 2
89
0 
9 12 10 1 5 7 6 3 8 2
12
0 
9 12 11 2 1 9 7 8 6 3
47
0 
10 9 3 8 6 12 11 2 4 5 10
35
0 
5 11 6 8 5 2
83
0 
3 12 9 11
92
0 
8 8 7 6 10 1 3 11 5
69
0 
4 8 6 7 11
42
0 
0 
84
0 
10 1 5 10 11 8 9 12 3 2 7
37
0 
0 
62
0 
1 12
40
0 
9 11 9 2 1 10 12 6 4 5
33
0 
10 5 10 3 2 6 9 8 11 1 12
94
0 
10 1 2 9 4 11 10 8 3 7 6
47
0 
2 2 10
77
0 
9 2 3 7 4 10 8 9 1 12
93
0 
5 8 12 3 9 5
49
0 
4 11 10 8 9
71
0 
1 7
82
0 
6 11 9 1 10 6 2
14
0 
9 6 3 10 2 12 9 8 4 1
85
0 
6 1 9 2 3 4 10
20
0 
10 3 12 9 11 6 1 4 2 5 8
66
0 
7 10 11 3 1 6 8 4
41
0 
5 12 11 3 9 6
72
0 
6 6 2 11 5 1 8
76
0 
0 
3
0 
5 2 6 7 3 9
65
0 
6 11 10 9 2 5 7
78
0 
8 11 6 5 3 12 4 8 7
62
0 
0 
82
0 
10 9 12 7 3 6 4 11 8 5 2
31
0 
2 12 6
56
0 
1 12
41
0 
1 5
51
0 
5 9 3 12 5 6
36
0 
7 2 6 7 12 1 8 11
52
0 
6 10 8 6 7 5 12
34
0 
8 12 11 7 5 3 1 2 8
51
0 
8 1 11 6 3 7 8 2 4
36
0 
7 6 10 7 3 11 5 2
34
0 
7 10 6 11 8 12 1 4
2
0 
2 1 8
26
0 
8 4 5 11 12 3 6 7 9
14
0 
2 12 5
58
0 
7 11 9 10 5 1 3 2
0
0 
6 6 5 7 2 10 4
86
0 
6 6 7 2 4 5 1
73
0 
7 10 7 8 4 11 2 6
19
0 
5 10 11 6 7 5
20
0 
5 7 8 3 10 6
60
0 
4 4 12 5 3
2
0 
5 12 10 7 8 3
16
0 
3 1 8 11
88
0 
10 2 8 3 12 10 4 7 1 9 5
27
0 
4 9 1 6 10
42
0 
6 9 5 4 12 10 7
16
0 
1 3
86
0 
8 1 11 9 8 6 10 4 2
37
0 
10 4 7 5 8 12 11 2 1 9 10
35
0 
1 5
98
0 
10 7 9 3 1 4 10 5 11 12 8
97
0 
10 4 10 12 1 5 6 8 9 7 2
83
0 
3 8 2 9
75
0 
4 10 9 2 12
30
0 
6 3 7 12 6 8 4
29
0 
5 11 10 8 9 6
31
0 
6 12 7 8 9 2 3
54
0 
10 12 7 6 5 10 8 3 2 9 4
84
0 
6 7 6 9 10 5 2
74
0 
7 3 1 2 4 6 9 10
9
0 
1 8
95
0 
8 5 9 12 11 10 2 8 7
9
0 
4 6 12 9 3
11
0 
0 
58
0 
5 5 7 6 12 8
44
0 
7 2 8 5 4 9 7 12
36
0 
8 8 9 4 10 12 1 7 5
66
0 
1 12
100
0 
2 11 1
92
0 
1 10
77
0 
8 5 3 11 12 4 1 10 7
94
0 
1 2
27
0 
5 7 6 8 4 12
44
0 
10 2 10 11 5 9 1 4 7 8 12
15
0 
3 3 1 12
12
0 
1 3
93
0 
10 9 8 3 2 1 5 12 10 6 11
70
0 
10 7 12 9 3 4 11 8 6 1 5
87
0 
3 5 1 6
49
0 
5 12 1 6 2 3
72
0 
8 9 3 11 8 6 2 4 7
40
0 
1 6
34
0 
10 3 7 8 6 5 1 12 4 10 11
65
0 
2 9 5
43
0 
9 6 3 7 8 4 12 1 10 2
67
0 
5 11 3 7 2 10
53
0 
5 11 9 7 1 2
71
0 
4 7 8 3 1
68
0 
0 
72
0 
8 7 10 6 12 5 8 3 9
90
0 
2 6 2
56
0 
2 3 1
77
0 
2 2 10
69
0 
9 4 6 11 8 5 1 12 3 10
77
0 
3 9 7 2
36
0 
8 11 1 9 6 7 5 4 8
79
0 
8 9 7 5 11 10 1 8 2
33
0 
10 2 3 8 12 9 7 10 5 1 11
96
0 
8 5 8 11 2 9 3 6 7
6
0 
1 9
74
0 
2 8 7
60
0 
2 2 5
50
0 
9 9 2 6 11 7 12 3 10 8
66
0 
5 3 4 5 6 7
74
0 
9 9 6 5 12 1 3 11 2 4
78
0 
10 2 4 9 6 10 8 3 1 11 7
37
0 
4 12 10 4 5
14
0 
10 7 12 8 3 10 6 5 11 4 1
56
0 
9 6 4 3 12 11 10 5 8 9
0
0 
7 9 2 6 3 1 11 4
68
0 
9 3 11 6 5 4 8 9 12 10
41
0 
8 6 5 1 10 12 11 8 2
87
0 
1 4
93
0 
7 3 6 1 12 5 4 8
88
0 
9 7 1 12 4 2 9 10 6 11
11
0 
8 12 7 6 8 1 11 4 3
0
0 
8 2 12 5 10 3 9 7 4
96
0 
6 6 10 11 2 3 5
96
0 
6 7 4 6 3 2 9
72
0 
10 9 1 4 10 12 3 5 2 11 7
39
0 
7 9 11 5 6 2 10 3
92
0 
1 3
94
0 
6 6 10 1 4 2 12
83
0 
1 9
87
0 
8 12 6 5 1 2 8 11 7
91
0 
10 10 1 9 3 2 5 11 7 12 8
5
0 
7 12 11 6 8 10 1 5
61
0 
7 7 4 9 10 6 2 11
7
0 
1 6
75
0 
3 9 2 10
31
0 
3 2 5 12
3
0 
6 11 10 7 12 3 4
45
0 
10 4 5 1 8 12 7 9 2 11 6
85
0 
5 10 4 6 1 5
37
0 
3 9 2 12
6
0 
9 11 12 4 10 3 6 1 2 5
93
0 
3 7 1 11
13
0 
10 9 10 11 5 3 12 7 1 4 8
4
0 
8 7 10 12 11 5 4 3 1
96
0 
5 5 6 2 3 1
25
0 
9 2 4 11 1 10 6 5 7 12
87
0 
4 2 12 9 7
13
0 
7 10 6 2 8 9 11 12
11
0 
9 1 10 4 7 11 9 2 3 12
42
0 
4 8 6 9 2
91
0 
4 7 9 12 11
42
0 
9 9 6 7 12 1 11 2 10 8
52
0 
6 10 3 7 1 8 11
22
0 
1 11
9
0 
9 11 9 7 4 8 5 12 10 3
97
0 
2 9 5
39
0 
4 9 4 6 12
20
0 
0 
87
0 
8 12 4 2 7 3 6 10 11
16
0 
1 8
48
0 
6 1 5 11 9 6 2
72
0 
6 10 5 4 2 8 3
46
0 
7 1 9 10 6 5 3 12
46
0 
9 10 2 11 1 12 4 3 8 7
67
0 
7 9 5 10 12 11 7 1
98
0 
6 3 5 1 6 2 12
47
0 
5 8 9 4 12 11
6
0 
3 8 1 4
90
0 
3 3 5 12
4
0 
9 5 6 10 11 1 12 2 7 4
15
0 
9 7 4 5 12 2 11 9 8 10
78
0 
7 10 8 7 6 12 1 9
57
0 
0 
35
0 
10 4 2 11 12 9 7 8 1 3 5
48
0 
9 11 2 7 6 4 12 3 5 10
76
0 
4 4 7 3 11
95
0 
1 3
53
0 
8 4 8 5 10 2 6 1 7
4
0 
2 4 11
5
0 
2 5 7
70
0 
4 9 5 6 2
70
0 
8 11 5 6 2 4 9 12 10
0
0 
10 3 11 5 6 4 1 8 7 10 2
75
0 
6 10 1 12 4 5 7
34
0 
4 10 1 4 5
56
0 
8 10 6 8 12 11 1 4 9
23
0 
1 5
40
0 
5 3 4 2 9 6
80
0 
10 9 2 6 1 3 11 4 7 10 8
3
0 
1 11
75
0 
1 9
22
0 
2 1 3
79
0 
8 5 12 4 9 7 6 3 11
13
0 
6 7 9 12 3 4 5
29
0 
6 6 4 5 2 1 9
94
0 
2 4 3
58
0 
0 
34
0 
9 1 7 9 3 11 10 2 8 6
70
0 
10 8 2 7 11 1 4 5 9 3 12
55
0 
2 8 5
36
0 
3 10 2 6
48
0 
1 8
87
0 
3 8 5 12
13
0 
7 7 8 11 1 12 4 10
16
0 
10 4 10 11 9 7 8 6 3 2 5
27
0 
2 5 9
55
0 
4 7 11 8 12
62
0 
1 6
16
0 
5 10 1 11 8 3
75
0 
8 1 3 5 4 6 12 2 11
34
0 
6 2 7 10 1 8 4
45
0 
9 5 12 2 4 8 11 6 10 1
77
0 
3 9 8 1
24
0 
8 6 5 9 8 3 7 1 11
34
0 
5 9 7 5 10 1
96
0 
0 
38
0 
0 
48
0 
1 11
53
0 
9 5 12 7 4 1 2 9 6 8
38
0 
9 11 3 9 1 2 5 6 12 8
20
0 
10 10 2 11 12 7 9 5 8 6 1
61
0 
2 3 10
80
0 
2 2 8
39
0 
5 3 8 4 7 1
21
0 
9 12 1 2 6 3 7 4 11 8
79
0 
6 2 11 12 5 8 7
87
0 
9 8 9 5 11 1 2 12 10 7
29
0 
0 
28
0 
10 7 5 3 2 10 8 9 6 12 1
74
0 
2 11 6
14
0 
3 2 8 3
25
0 
9 8 5 7 2 6 9 12 1 3
64
0 
7 3 5 6 11 7 12 1
91
0 
1 9
29
0 
7 5 7 10 1 9 8 6
46
0 
7 1 8 11 5 4 9 3
65
0 
6 2 11 8 12 3 7
68
0 
6 3 9 4 2 6 1
13
0 
1 8
2
0 
6 8 12 7 11 3 9
7
0 
7 3 4 10 2 9 8 6
33
0 
2 2 1
30
0 
3 10 4 9
75
0 
0 
28
0 
9 6 12 8 11 4 9 2 7 5
86
0 
5 2 9 10 1 4
4
0 
9 10 7 12 4 11 1 2 9 5
27
0 
1 8
95
0 
10 2 7 10 5 4 3 1 8 12 11
88
0 
1 3
29
0 
2 1 2
22
0 
2 12 3
8
0 
8 6 2 7 12 1 8 3 9
54
0 
0 
89
0 
6 3 10 11 6 12 2
30
0 
1 6
58
0 
10 11 5 2 7 4 3 10 1 8 12
38
0 
5 6 9 8 2 5
49
0 
5 5 10 1 4 6
100
0 
8 3 1 11 2 12 9 6 4
20
0 
9 2 8 7 12 4 1 5 10 6
67
0 
3 8 4 11
59
0 
9 1 9 4 6 12 8 11 5 7
85
0 
3 4 6 7
32
0 
9 6 12 4 5 10 2 9 1 3
29
0 
7 5 4 9 8 10 1 7
14
0 
3 7 9 5
34
0 
7 12 10 1 4 11 3 5
93
0 
1 5
10
0 
8 12 11 4 1 3 9 5 10
10
0 
4 5 3 11 9
93
0 
2 9 10
9
0 
2 7 11
25
0 
2 8 10
40
0 
2 10 4
73
0 
3 10 7 1
98
0 
9 1 4 6 8 7 2 3 10 9
14
0 
2 7 4
9
0 
3 3 6 2
6
0 
4 7 4 1 8
61
0 
4 4 3 9 5
30
0 
10 4 1 5 11 6 2 7 3 8 10
23
0 
1 10
83
0 
5 5 3 10 12 8
4
0 
4 8 9 4 12
84
0 
1 2
67
0 
6 5 4 3 10 2 1
67
0 
2 9 6
9
0 
8 5 8 4 2 11 6 12 3
60
0 
10 1 8 2 3 12 4 5 9 7 11
79
0 
3 3 10 6
35
0 
6 3 1 7 2 4 8
41
0 
0 
43
0 
7 5 6 2 7 1 8 12
3
0 
3 4 1 5
47
0 
3 6 2 8
34
0 
4 9 7 1 10
62
0 
5 6 1 9 12 7
69
0 
2 2 11
38
0 
10 1 5 8 4 12 6 10 11 9 2
60
0 
1 4
13
0 
7 1 8 5 4 3 12 6
8
0 
6 11 8 9 6 4 3
20
0 
2 4 11
34
0 
3 7 5 2
21
0 
10 5 2 10 4 7 6 3 8 12 9
24
0 
5 12 3 11 10 9
0
0 
2 6 10
65
0 
3 3 9 4
18
0 
3 12 5 3
39
0 
9 10 3 8 7 5 2 6 11 9
64
0 
9 2 11 7 6 9 12 5 10 1
61
0 
1 9
74
0 
0 
13
0 
5 5 12 4 3 10
61
0 
10 11 1 6 12 8 10 2 3 4 9
81
0 
6 8 2 1 12 6 7
99
0 
8 11 6 12 3 4 1 2 10
3
0 
7 3 7 12 2 11 8 4
22
0 
9 5 9 6 7 2 8 12 10 11
20
0 
1 3
68
0 
1 5
55
0 
9 12 7 4 1 6 10 3 2 5
51
0 
5 2 4 11 6 8
61
0 
5 11 4 3 8 5
44
0 
1 2
30
0 
0 
68
0 
8 8 12 1 10 9 11 7 2
11
0 
8 1 8 3 7 12 5 2 6
95
0 
9 12 9 1 3 6 2 5 10 4
1
0 
8 2 7 4 9 11 12 6 10
0
0 
1 6
13
0 
9 3 8 12 5 10 1 9 2 6
78
0 
2 2 6
24
0 
9 4 6 12 3 8 1 9 2 5
30
0 
1 3
57
0 
1 8
63
0 
2 4 1
44
0 
3 7 4 5
68
0 
5 1 9 2 11 5
52
0 
9 7 4 10 8 6 11 9 1 3
68
0 
7 8 12 7 3 4 10 11
63
0 
10 5 8 4 6 11 10 7 12 1 2
73
0 
4 8 2 9 4
70
0 
5 1 10 11 3 6
33
0 
0 
19
0 
6 9 11 12 2 6 8
90
0 
2 9 2
62
0 
0 
68
0 
0 
47
0 
9 8 3 1 6 10 11 4 9 12
1
0 
8 7 5 2 10 12 9 11 6
1
0 
5 4 9 12 5 6
87
0 
3 8 2 7
79
0 
1 12
76
0 
2 12 3
1
0 
0 
85
0 
1 9
85
0 
10 11 3 6 1 8 7 5 12 10 9
52
0 
0 
85
0 
1 11
99
0 
0 
62
