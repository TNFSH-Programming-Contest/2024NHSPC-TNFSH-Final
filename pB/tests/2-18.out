80
18829422
176
112
1394
38
162
34
46
4393339
101
135369
85
18828191
23221562
18965010
54
139
135386
177415264
100
18828215
84
87
12
4258016
111
56
145
139
135478
62
65
135467
177551898
3
82
66
161
18828234
135442
176
18963693
58
77
23087369
177415223
35
196243392
18963637
72
128
135432
178
177552054
1401
200502657
181809804
1368
1475
76
156
135439
12
87
135594
177550703
4259298
35
177415207
23087473
118
78
34
17
4259451
38
28
18
200501324
112
28
59
1495
88
1357
92
43
1288
75
13
133
68
14
1351
19
47
85
85
62
